### PR TITLE
Add documentation for hooks.

### PR DESCRIPTION
**Pull request by kyrofa**
Finish off LP: [#1586465](https://bugs.launchpad.net/ubuntu/+source/snapd/+bug/1586465) by documenting how to use snapcraft's hook support.

### CODE REVIEW

File: docs/hooks.md [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+use the top-level attribute `hooks` in `snapcraft.yaml` to request plugs, like
+so:
+
+    hooks: # Top-level YAML attribute, parallel to `apps`
+        configure: # Hook name, corresponds to executable name
+            plugs: [network] # Or any other plugs required by this hook
+
+Note that hooks will be called with no parameters. If they need more information
+from snapd (or need to provide information to snapd) they can utilize the
+`snapctl` command (for more information on `snapctl`, see `snapctl -h`).
```
Comment (elopio) [inline docs/hooks.md]: s/see/run

File: docs/hooks.md [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+As long as the file name of the executable corresponds to a hook name supported
+by snapd, that's all one needs to do in order to utilize a hook within their
+snap. Note that hooks, like apps, are executed within a confined environment. By
+default hooks will run with no plugs; if a hook needs more privileges one can
+use the top-level attribute `hooks` in `snapcraft.yaml` to request plugs, like
+so:
+
+    hooks: # Top-level YAML attribute, parallel to `apps`
+        configure: # Hook name, corresponds to executable name
+            plugs: [network] # Or any other plugs required by this hook
```
Comment (elopio) [inline docs/hooks.md]: we kind of have the convention to use two spaces for indentation in the yaml. I would prefer to have two instead of four here.

Comment (kyrofa) [inline docs/hooks.md]: Yeah good point, done.

File: docs/hooks.md (hunk 1 above)
Comment (kyrofa) [inline docs/hooks.md]: Done.

File: docs/hooks.md [hunk 3]
```diff
@@ -0,0 +1,34 @@
+# Snapcraft hook support
+
+Snapcraft supports snapd's [hooks][1] mechanism with two conventions:
```
Comment (caldav) [inline docs/hooks.md]: s/"with two conventions"/"using two possible methods"
Convention feels mandatory and makes the numbered list look like a list of steps.

Comment (kyrofa) [inline docs/hooks.md]: Done, thank you!

**Pull request merged**